generated/nidaqmx/_library_interpreter.py

```diff
@@ -2230,11 +2230,49 @@ def get_ai_chan_cal_exp_date(self, task, channel_name):
 
     def get_analog_power_up_states(
             self, device_name, channel_name, channel_type):
-        raise NotImplementedError
+        state = []
+
+        args = [device_name]
+        argtypes = [ctypes_byte_str]
+
+        for index in range(len(channel_name)):
+            state_element = ctypes.c_double
+            state.append(state_element)
+
+            args.append(channel_name[index])
+            argtypes.append(ctypes_byte_str)
+            
+            args.append(ctypes.byref(state_element))
+            argtypes.append(ctypes.POINTER(ctypes.c_double))
+            
+            args.append(channel_type[index])
+            argtypes.append(ctypes.c_int32)
+            
+        args.append(None)
+
+        cfunc = lib_importer.cdll.DAQmxGetAnalogPowerUpStates
+        with cfunc.arglock:
+            cfunc.argtypes = argtypes
+            error_code = cfunc(*args)
+        check_for_error(error_code)
+        return state.tolist()
```

Comment:
Isn't this returning a list of `ctypes.c_double`? Should we be unwrapping the values in there instead?

Reply:
Yes, this is mixing types and values.

`ctypes.c_double` is a type. You should put it in `argtypes`.

`ctypes.c_double()` is an instance of `ctypes.c_double`. You should put it in args using `ctypes.byref()`.

 `ctypes.c_double().value` is a Python float. You should return a list of these from the function.

Reply:
Yes and here the `state` variable is a list, hence i am trying to return a list in this implementation.

Reply:
Right, but `state` is currently a list of types, not objects of a type. Even if it were objects of a type, it is the wrong type. You want something like this:

```python
state_list = []
for chans:
   state = ctypes.c_double() # note ctor call
   state_list.append(state)
   args.append(ctypes.byref(state))

return [state.value for state in state_list]
```

Reply:
Thanks for the clarification, i have updated as per suggestions